doc/dask.rst

```diff
@@ -382,6 +391,95 @@ application.
     structure of a problem, unlike the generic speedups offered by
     ``dask='parallelized'``.
 
+
```

Comment:
@aaronspring @bradyrx @andersy005 As likely downstream users of map_blocks, can you read through this doc file and see if it is clear and understandable?